hxd/fs/FileEntry.hx

```diff
@@ -1,34 +1,142 @@
+	/**
+		Returns the first four bytes of the file which usually is a file signature.
+	**/
 	public function getSign() : Int return 0;
 
+	/**
+		Returns the contents of the file as Bytes.
+	**/
 	public function getBytes() : haxe.io.Bytes return null;
 
+	/**
+		Returns the contents of the file as a String.
+	**/
 	public function getText() return getBytes().toString();
 
+	/**
+		Prepares file to be read from the start or moves the reading position to the beginning if its already opened.
+
+		This is an alternative way to read file data without storing its entirety in the RAM.
+		Note that depending on FileSystem used, they always load entire file into RAM, such as EmbedFileSystem or BytesFileSystem.
+		However for LocalFileSystem this is a preferred method of reading files when entirety of the file is not required in the RAM.
+
+		Required to be called before `FileEntry.skip`, `FileEntry.readByte` and `FileEntry.read` can be called.
+		`FileEntry.close` should be called after the file reading is complete.
+	**/
 	public function open() { }
+	/**
+		Moves the reading position of the file by `nbytes`.
+
+		File have to be opened first via `FileEntry.open`.
+	**/
 	public function skip( nbytes : Int ) { }
+	/**
+		Reads a single byte from the file.
+
+		File have to be opened first via `FileEntry.open`.
+	**/
 	public function readByte() : Int return 0;
+	/**
+		Reads the bytes from the file into the output Bytes.
+
+		File have to be opened first via `FileEntry.open`.
+
+		@param out The output Bytes instance file data is written into.
+		@param pos The position in `out` starting from which the data is written.
+		@param size The amount of bytes to read.
+	**/
 	public function read( out : haxe.io.Bytes, pos : Int, size : Int ) {}
+	/**
+		Closes the file previously opened file via `FileEntry.open`.
+	**/
 	public function close() {}
 
+	/**
+		Loads the file, preparing it to be accessed. `FileEntry.isAvailable` can be used to determine is file is loaded/ready to be accessed.
+
+		This is an asynchronous operation and delayed for at least one frame. Only exception is `LocalFileSystem` in macro context.
+		
+		@param onReady The callback that is invoked after the file is loaded.
+	**/
 	public function load( ?onReady : Void -> Void ) : Void { if( !isAvailable ) throw "load() not implemented"; else if( onReady != null ) onReady(); }
+	/**
+		Loads the file as platform-specific Bitmap. Will throw an exception for anything except JS and flash targets.
+		For FileSystem-independent bitmap loading use `hxd.res.Image` Resource.
+
+		This is an asynchronous operation and delayed for at least one frame.
+
+		@param onLoaded The callback that is invoked after the bitmap is loaded.
+	**/
 	public function loadBitmap( onLoaded : LoadedBitmap -> Void ) : Void { throw "loadBitmap() not implemented"; }
+	/**
+		Set the file change callback to the file. Only one callback can be set at a time and only works for `LocalFileSystem`.

```

Comment:
Register a function that will be called if the file changes. This might only be supported on file based filesystems.